src/core/server/legacy/legacy_internals.test.ts

```diff
@@ -17,36 +17,16 @@
  * under the License.
  */
 
-import { Server } from 'hapi';
-
-import { configMock } from '../config/mocks';
-import { httpServiceMock } from '../http/http_service.mock';
 import { httpServerMock } from '../http/http_server.mocks';
-import { findLegacyPluginSpecsMock } from './legacy_service.test.mocks';
 import { LegacyInternals } from './legacy_internals';
-import { ILegacyInternals, LegacyConfig, LegacyVars, LegacyUiExports } from './types';
-
-function varsProvider(vars: LegacyVars, configValue?: any) {
-  return {
-    fn: jest.fn().mockReturnValue(vars),
-    pluginSpec: {
-      readConfigValue: jest.fn().mockReturnValue(configValue),
-    },
-  };
-}
+import { ILegacyInternals } from './types';
 
 describe('LegacyInternals', () => {
   describe('getInjectedUiAppVars()', () => {
-    let uiExports: LegacyUiExports;
-    let config: LegacyConfig;
-    let server: Server;
     let legacyInternals: ILegacyInternals;
 
     beforeEach(async () => {
-      uiExports = findLegacyPluginSpecsMock().uiExports;
-      config = configMock.create() as any;
-      server = httpServiceMock.createInternalSetupContract().server;
-      legacyInternals = new LegacyInternals(uiExports, config, server);
+      legacyInternals = new LegacyInternals();
```

Comment:
I couldn't get rid of the `LegacyInternals` class, that is still used by the rendering service. I guess it will have to wait until we totally remove the legacy server.

Reply:
Done in https://github.com/elastic/kibana/pull/77599/commits/7e4a35bc0987c4ee87fdad87bbb851d88f31a343